### PR TITLE
OEL-2743: Theme views table.

### LINKED ISSUE #312: Theme views table
Recently you have themed the tables https://github.com/openeuropa/oe_bootstrap_theme/pull/290 using BCL. Thanks!
It would be nice to also consider theming the views table.

### PR DESCRIPTION
**Pull request by AaronGilMartinez**
Jira issue(s):
- [D8AGE-???](https://webgate.ec.europa.eu/CITnet/jira/browse/D8AGE-???)

### CODE REVIEW

File: oe_bootstrap_theme.theme [hunk 1]
```diff
@@ -56,3 +56,29 @@ function oe_bootstrap_theme_preprocess_table__bootstrap__responsive(&$variables)
 
   $variables['wrapper_attributes'] = new Attribute(['class' => $class]);
 }
+
+/**
+ * Implements hook_preprocess_HOOK() for views-view-table.html.twig.
+ */
+function oe_bootstrap_theme_preprocess_views_view_table(&$variables) {
```
Comment (drishu) [inline oe_bootstrap_theme.theme]: Let's go with the same approach as we did with the table templates:

- implement `oe_bootstrap_theme_theme_suggestions_views_view_table_alter`
- there we add `views_view_table__bootstrap` and `views_view_table__bootstrap__responsive`.
- then  implement `oe_bootstrap_theme_preprocess_views_view_table__bootstrap__responsive`
- template `views-view-table--bootstrap.html.twig`
- and `views-view-table--bootstrap--responsive.html.twig` extending the first one

File: tests/src/Kernel/ViewsTableTest.php [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    'views_test_data',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Configuration and schemas needed for view test.
```
Comment (drishu) [inline tests/src/Kernel/ViewsTableTest.php]: instead of this comment and the one from line 38, rather explain why we do all this code ex: `// Here we set up tralalal` and leave a @see reference to ViewsKernelTestBase::setUpFixtures()

File: tests/src/Kernel/ViewsTableTest.php [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    $query->execute();
+
+    // Enable the test table view.
+    $storage = \Drupal::entityTypeManager()->getStorage('view');
+    $config_dir = \Drupal::service('extension.list.module')->getPath('views_test_config') . '/test_views';
+    $file_storage = new FileStorage($config_dir);
+    $storage->create($file_storage->read('views.view.test_table'))->save();
+
+    // Enable the theme.
+    $this->container->get('theme_installer')->install(['oe_bootstrap_theme']);
```
Comment (drishu) [inline tests/src/Kernel/ViewsTableTest.php]: Isn't the theme already enabled in the parent::setUp ?

**Review by drishu — CHANGES_REQUESTED** [review]

File: tests/src/Kernel/ViewsTableTest.php (hunk 3 above)
Comment (AaronGilMartinez) [reply tests/src/Kernel/ViewsTableTest.php]: Indeed, I was reinstalling system configuration and causing issues.

File: tests/src/Kernel/ViewsTableTest.php (hunk 2 above)
Comment (AaronGilMartinez) [reply tests/src/Kernel/ViewsTableTest.php]: Ok, this way explains why the code it's there and that I'm following core

File: oe_bootstrap_theme.theme (hunk 1 above)
Comment (AaronGilMartinez) [reply oe_bootstrap_theme.theme]: Thanks, this saved time

File: oe_bootstrap_theme.theme [hunk 4]
```diff
@@ -48,6 +68,20 @@ function oe_bootstrap_theme_preprocess(&$variables) {
...
+
+/**
+ * Implements hook_preprocess_HOOK() for views-view-table--bootstrap--responsive.html.twig.
+ */
+function oe_bootstrap_theme_preprocess_views_view_table__bootstrap__responsive(&$variables) {
+  _oe_bootstrap_theme_bootstrap_responsive_table($variables);
+}
+
+/**
+ * Function that applies responsiveness to Bootstrap tables.
```
Comment (drishu) [inline oe_bootstrap_theme.theme:82]: Since this is not a drupal hook you need to add parameters

File: oe_bootstrap_theme.theme [hunk 5]
```diff
@@ -48,6 +68,20 @@ function oe_bootstrap_theme_preprocess(&$variables) {
...
+ */
+function _oe_bootstrap_theme_bootstrap_responsive_table(&$variables) {
```
Comment (drishu) [inline oe_bootstrap_theme.theme]: add return type

File: templates/patterns/offcanvas/pattern-offcanvas.html.twig [hunk 6]
```diff
@@ -10,8 +10,8 @@
   'placement': placement,
   'id': offcanvas_id,
   'body': body,
-  'with_body_scroll': body_scroll,
-  'with_backdrop': backdrop,
+  'with_body_scroll': true,
+  'with_backdrop': true,
```
Comment (drishu) [inline templates/patterns/offcanvas/pattern-offcanvas.html.twig]: why these changes? seems unrelated and these variables exist in the pattern definition so its not ok to change them

**Review by drishu — CHANGES_REQUESTED** [review]

File: oe_bootstrap_theme.theme (hunk 4 above)
Comment (AaronGilMartinez) [reply oe_bootstrap_theme.theme:82]: True

File: oe_bootstrap_theme.theme (hunk 5 above)
Comment (AaronGilMartinez) [reply oe_bootstrap_theme.theme]: Ok

File: templates/patterns/offcanvas/pattern-offcanvas.html.twig (hunk 6 above)
Comment (AaronGilMartinez) [reply templates/patterns/offcanvas/pattern-offcanvas.html.twig]: Added by mistake

**Review by drishu — APPROVED** [review]

**Review by intelektron — APPROVED** [review]
